src/arrayMethodJoin.js

```diff
@@ -4,8 +4,41 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let addToString = '';
+    let sep = separator;
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        if (this[i] === undefined) {
+          result = result + '';
+
+          return result;
+        }
```

Comment:
this if statement will be redundant if you change logic a bit: add separator only if it's not the last element

Reply:
and in this case, `return` inside the loop will be redundant

Reply:
I can't figure out what to do (
Tell me, please.

Reply:
why do you need this statement, if you add just empty string it's mean that you didn't add anything

Reply:
you need just two checks:
1. if current element isn't null and isn't undefined, you shoud add it to resulted string
2. if current element is not last, add separator to your resulted string